src/item_factory.cpp

```diff
@@ -634,6 +634,25 @@ void Item_factory::load( islot_gun &slot, JsonObject &jo )
                                              curr.get_int( 1 ) ) );
         }
     }
+
+    //Add any built-in mods.
+    if( jo.has_array( "built_in_mods" ) ) {
```

Comment:
This (and the other setting below) should be documented in doc/JSON_INFO.md, under the `"type": "GUN"` section.

And you could add code in `Item_factory::check_definitions` to check that each entry of those vectors contains valid item ids. This catches spelling mistakes when the item data is loaded.